File: Level 2/프로세스/solution.js
function solution(priorities, location) {
  priorities = priorities.map((priority, index) => ({
    target: location === index,
    priority,
  }));

  let count = 0;

  while (priorities.length !== 0) {
    const shiftedPriority = priorities.shift();
    const isExistHigherPriority = priorities.some(
      (priority) => priority.priority > shiftedPriority.priority
    );

    if (isExistHigherPriority) {
      priorities.push(shiftedPriority);
      continue;
    }

    count++;

    if (shiftedPriority.target) {
      return count;
    }
  }
}

File: Level 2/프로세스/solution.py
from collections import deque


def solution(priorities, location):
    queue = deque((index, priority) for index, priority in enumerate(priorities))

    answer = 1

    while queue:
        poped_index, poped_priority = queue.popleft()

        if any(priority > poped_priority for _, priority in queue):
            queue.append((poped_index, poped_priority))
            continue
        else:
            if poped_index == location:
                return answer
            else:
                answer += 1
